flaml/model.py

```diff
@@ -186,6 +187,13 @@ def _preprocess(self, X):
             X = X.copy()
             cat_columns = X.select_dtypes(include=['category']).columns
             X[cat_columns] = X[cat_columns].apply(lambda x: x.cat.codes)
+        elif isinstance(X, np.ndarray) and X.dtype.kind not in 'buif':

```

Comment:
looks fine as a fix - although, it is not clear why string type is a problem when X is numpy array and works fine when X originally is DataFrame type, and even the fix first converts the input to DataFrame type

Reply:
when X is a DataFrame, it also needs to be processed. See the lines above.

Reply:
I guess the main question was -- wouldn't it be simpler to just convert numpy array to dataframe and then let the original processing for dataframe take over, rather than having a separate special fix for np array. If there are reasons not to do this - this change is fine.